assets/src/components/ladder.tsx

```diff
@@ -1,56 +1,60 @@
-import React, { useContext } from "react"
-import DispatchContext from "../contexts/dispatchContext"
+import React from "react"
+import {
+  LadderVehiclePosition,
+  putIntoLanes,
+  VehicleDirection,
+  vehicleDirectionOnLadder,
+  xInLaneWithOffset,
+} from "../models/ladderVehiclePosition"
 import { Timepoint, Vehicle, VehicleId, VehicleTimepointStatus } from "../skate"
-import { selectVehicle } from "../state"
-import VehicleIcon from "./vehicleIcon"
+import LadderVehicle, {
+  centerOfVehicleGroupX,
+  widthOfVehicleGroup,
+} from "./ladderVehicle"
+
+export interface Props {
+  timepoints: Timepoint[]
+  vehicles: Vehicle[]
+  ladderDirection: LadderDirection
+  selectedVehicleId: VehicleId | undefined
+}
 
 // Timepoints come from the API in the ZeroToOne direction
 export enum LadderDirection {
   ZeroToOne,
   OneToZero,
 }
 
-export enum VehicleDirection {
-  Up,
-  Down,
-}
-
-export const vehicleDirectionOnLadder = (
-  vehicle: Vehicle,
-  ladderDirection: LadderDirection
-): VehicleDirection =>
-  (vehicle.direction_id === 1) ===
-  (ladderDirection === LadderDirection.ZeroToOne)
-    ? VehicleDirection.Down
-    : VehicleDirection.Up
-
 export const flipLadderDirection = (
   ladderDirection: LadderDirection
 ): LadderDirection =>
   ladderDirection === LadderDirection.ZeroToOne
     ? LadderDirection.OneToZero
     : LadderDirection.ZeroToOne
 
+const occupiedLanes = (ladderVehiclePositions: LadderVehiclePosition[]) => {
+  const allLanes: number[] = ladderVehiclePositions.map(
+    ({ lane }: LadderVehiclePosition) => lane || 0
+  )
+  const furthestLane = allLanes.length ? Math.max(...allLanes) : 0
+  return furthestLane + 1
+}
+
+const ladderVehicleHorizontalOffsetForLanes = (maxOccupiedLane: number) => {
+  const cappedLanes = Math.min(maxOccupiedLane, 5)
+  return ((8 - cappedLanes) / 8) * widthOfVehicleGroup
```

Comment:
This is defining the offset for each lane based on the max number of vehicles wide this ladder gets in order to condense the vehicles more as there are more of them across. So, if there are 2 vehicles across it will space them at a wider 75% of the icon width, while at 5 vehicles across it'll be tighter at 37.5%. It is capped at that condensed to ensure you can still click on the edge of any icon.